app/models/sample.rb

```diff
@@ -6,9 +6,22 @@ class Sample < ApplicationRecord
 
   attr_readonly :name
 
+  enum retention_instruction: {
+    destroy_after_2_years: 0,
+    return_to_customer_after_2_years: 1,
+    long_term_storage: 2
+  }
+
+  VALID_RETENTION_INSTRUCTIONS = %w[
+    destroy_after_2_years
+    return_to_customer_after_2_years
+    long_term_storage
+  ].freeze
+
   validates :species, presence: true
   validates :external_id, presence: true, uuid: true
   validates :name, uniqueness: { case_sensitive: false }, presence: true
+  validates :retention_instruction, inclusion: { in: VALID_RETENTION_INSTRUCTIONS }, allow_nil: true
```

Comment:
You probably need some tests to go with this to see how it handles a message actually trying to set an invalid value.  I'm not entirely sure where this will break.  The whole point of an enum is that you cannot set invalid values so it may break before this validation even occurs.